src/utils/randomId.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Creates a random URL-safe string for IDs and unique fallback values.
+ *
+ * @param len Length of the string to generate.
+ */
+export default function randomId(len: number = 21): string {
+	const charset = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-';
```

Comment:
Same default length (21) and alphabet (`A-Za-z0-9_-`) as [nanoid](https://www.npmjs.com/package/nanoid).